taskcluster/ci/train-vocab/kind.yml

```diff
@@ -0,0 +1,90 @@
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at http://mozilla.org/MPL/2.0/.
+---
+
+loader: taskgraph.loader.transform:loader
+
+transforms:
+    - translations_taskgraph.transforms.from_datasets:locales_only
+    - translations_taskgraph.transforms.command_context_from_params:transforms
+    - taskgraph.transforms.job:transforms
+    - translations_taskgraph.transforms.cache:transforms
+    - taskgraph.transforms.cached_tasks:transforms
+    - taskgraph.transforms.task:transforms
+
+kind-dependencies:
+    - merge-corpus
+    - toolchain
+
+tasks:
+    "{src_locale}-{trg_locale}":
+        description: train vocab for {src_locale}-{trg_locale}
+        attributes:
+            # TODO: is this `train` or `devset`? need to decode snakemake a bit more..
+            dataset-category: train
+            stage: train-vocab
+            cache:
+                type: train-vocab
+                resources:
+                    - pipeline/train/spm-vocab.sh
+                parameters:
+                    - train_vocab_sample_size
+        dataset-config:
+            substitution-fields:
+                - description
+                - name
+                - treeherder.symbol
+                - fetches
+                - dependencies
+        worker-type: b-linux-large
+        worker:
+            docker-image: {"in-tree": "train"}
+            max-run-time: 3600
+            artifacts:
+                - name: public/build
+                  path: /builds/worker/artifacts
+                  type: directory
+            env:
+                COMPRESSION_CMD: zstdmt
+
+        # Don't run unless explicitly scheduled
+        run-on-tasks-for: []
+
+        treeherder:
+            symbol: "{src_locale}-{trg_locale}"
+            platform: train-vocab/opt
+        run:
+            using: run-task
+            command-context:
+                from-parameters:
+                    - train_vocab_sample_size
+            command:
+                - bash
+                - -c
+                # Arguments are:
+                # 1) merged src corpus file
+                # 2) merged trg corpus file
+                # 3) output file
+                # 4) sample size
+                # 5) number of threads (auto = output of nproc)
+                - >-
+                    export MARIAN=$MOZ_FETCHES_DIR &&
+                    $VCS_PATH/pipeline/train/spm-vocab.sh
+                    fetches/corpus.{src_locale}.zst
+                    fetches/corpus.{trg_locale}.zst
+                    artifacts/vocab.spm
```

Comment:
Why not `artifacts/<something>/vocab.spm`?

Reply:
We could do that, yeah, although it might be a bit annoying for downstream steps (we'd end up with an extra directory in the artifact path). I'm good either way.

Reply:
nah, I see the prefix above is not a directory